aws/service_v2.go

```diff
@@ -131,6 +157,106 @@ func DynamoDbClient(ctx context.Context, d *plugin.QueryData) (*dynamodb.Client,
 	return svc, nil
 }
 
+// Ec2Client returns the service client for AWS Ec2 service
+func Ec2Client(ctx context.Context, d *plugin.QueryData) (*ec2.Client, error) {
+	region := d.KeyColumnQualString(matrixKeyRegion)
+	if region == "" {
+		return nil, fmt.Errorf("region must be passed DynamodbClient Client")
+	}
+	// have we already created and cached the service?
+	serviceCacheKey := fmt.Sprintf("ec2-v2-%s", region)
+	if cachedData, ok := d.ConnectionManager.Cache.Get(serviceCacheKey); ok {
+		return cachedData.(*ec2.Client), nil
+	}
+
+	// so it was not in cache - create service
+	cfg, err := getSessionV2(ctx, d, region)
+	if err != nil {
+		plugin.Logger(ctx).Error("Ec2Client", "service_client_error")
+		return nil, err
+	}
+
+	svc := ec2.NewFromConfig(*cfg)
+	d.ConnectionManager.Cache.Set(serviceCacheKey, svc)
+
+	return svc, nil
+}
+
+// ELBv2Client returns the service client for AWS ElasticLoadBalance service
+func ELBv2Client(ctx context.Context, d *plugin.QueryData) (*elbv2.Client, error) {
+	region := d.KeyColumnQualString(matrixKeyRegion)
+	if region == "" {
+		return nil, fmt.Errorf("region must be passed DynamodbClient Client")
+	}
+	// have we already created and cached the service?
+	serviceCacheKey := fmt.Sprintf("elbv2-v2-%s", region)
+	if cachedData, ok := d.ConnectionManager.Cache.Get(serviceCacheKey); ok {
+		return cachedData.(*elbv2.Client), nil
+	}
+
+	// so it was not in cache - create service
+	cfg, err := getSessionV2(ctx, d, region)
+	if err != nil {
+		plugin.Logger(ctx).Error("ELBv2Client", "service_client_error")
+		return nil, err
+	}
+
+	svc := elbv2.NewFromConfig(*cfg)
+	d.ConnectionManager.Cache.Set(serviceCacheKey, svc)
+
+	return svc, nil
+}
+
+// ELBClient returns the service client for AWS ElasticLoadBalance service
+func ELBClient(ctx context.Context, d *plugin.QueryData) (*elb.Client, error) {
+	region := d.KeyColumnQualString(matrixKeyRegion)
+	if region == "" {
+		return nil, fmt.Errorf("region must be passed DynamodbClient Client")
+	}
+	// have we already created and cached the service?
+	serviceCacheKey := fmt.Sprintf("elb-v2-%s", region)
+	if cachedData, ok := d.ConnectionManager.Cache.Get(serviceCacheKey); ok {
+		return cachedData.(*elb.Client), nil
+	}
+
+	// so it was not in cache - create service
+	cfg, err := getSessionV2(ctx, d, region)
+	if err != nil {
+		plugin.Logger(ctx).Error("ELBv2Client", "service_client_error")
+		return nil, err
+	}
+
+	svc := elb.NewFromConfig(*cfg)
+	d.ConnectionManager.Cache.Set(serviceCacheKey, svc)
+
+	return svc, nil
+}
+
+// AutoscalingClient returns the service client for AWS Autoscaling service
+func AutoscalingClient(ctx context.Context, d *plugin.QueryData) (*autoscaling.Client, error) {
```

Comment:
```suggestion
func AutoScalingClient(ctx context.Context, d *plugin.QueryData) (*autoscaling.Client, error) {
```

Please be careful when converting service names to camel case